ddtrace/ext/git.py

```diff
@@ -13,3 +27,85 @@
 
 # Git Tag
 TAG = "git.tag"
+
+# Git Commit Author Name
+COMMIT_AUTHOR_NAME = "git.commit.author.name"
+
+# Git Commit Author Email
+COMMIT_AUTHOR_EMAIL = "git.commit.author.email"
+
+# Git Commit Author Date (UTC)
+COMMIT_AUTHOR_DATE = "git.commit.author.date"
+
+# Git Commit Committer Name
+COMMIT_COMMITTER_NAME = "git.commit.committer.name"
+
+# Git Commit Commiter Email
+COMMIT_COMMITTER_EMAIL = "git.commit.committer.email"
+
+# Git Commit Committer Date (UTC)
+COMMIT_COMMITTER_DATE = "git.commit.committer.date"
+
+# Git Commit Message
+COMMIT_MESSAGE = "git.commit.message"
+
+log = logging.getLogger(__name__)
+
+
+def extract_user_info(author=True):
+    # type: (bool) -> Tuple[str, str, str]
+    """Extract git commit author/committer info."""
+    formatting = "--format=%an,%ae,%ad" if author else "--format=%cn,%ce,%cd"
+    cmd = subprocess.Popen(
+        ["git", "show", "-s", formatting, "--date=format:%Y-%m-%dT%H:%M:%S%z"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    stdout, stderr = cmd.communicate()
+    if cmd.returncode == 0:
+        name, email, date = compat.ensure_text(stdout).strip().split(",")
+        return name, email, date
+    raise ValueError(stderr)
+
+
+def extract_repository_url():
+    # type: () -> str
+    cmd = subprocess.Popen(["git", "ls-remote", "--get-url"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = cmd.communicate()
+    if cmd.returncode == 0:
+        repository_url = compat.ensure_text(stdout).strip()
+        return repository_url
+    raise ValueError(stderr)
+
+
+def extract_commit_message():
+    # type: () -> str
+    cmd = subprocess.Popen(["git", "show", "-s", "--format=%s"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = cmd.communicate()
+    if cmd.returncode == 0:
+        commit_message = compat.ensure_text(stdout).strip()
+        return commit_message
+    raise ValueError(stderr)
+
+
+def extract_git_metadata():
```

Comment:
is there any minimum git version requirement for any other these commands/options?

Reply:
Looks like `git show -s` has been supported at the very latest in version 2.1.4 (from 2014), while `git ls-remote` has been supported at the latest in version 2.6.7 (from 2017). Is this something we should be careful about?

Reply:
not sure, just wanted to be sure, maybe we can add comments here on what is supported

or in docs if we say "automatically get git info from .git ..... " we can also specify the minimum required `git` binary version.

what do you think?

Reply:
Where in the tracer docs do we talk about CI app? Just curious (or do you mean the CI apps docs themselves?)

Reply:
"yes" not sure I had a specific place in mind (or even that I know about 😆)

Reply:
Hmm... I'll look out for any relevant docs, but I think leaving the comments about versions may be sufficient for now. WDYT?

Reply:
correct, if there are no docs in this codebase, then comments as you have them are good!